File: app/src/main/java/dev/ragnarok/fenrir/domain/ILocalServerInteractor.java
package dev.ragnarok.fenrir.domain;

import java.util.List;

import dev.ragnarok.fenrir.model.Audio;
import dev.ragnarok.fenrir.model.Photo;
import dev.ragnarok.fenrir.model.Video;
import io.reactivex.rxjava3.core.Single;

public interface ILocalServerInteractor {
    Single<List<Video>> getVideos(int offset, int count, boolean reverse);

    Single<List<Audio>> getAudios(int offset, int count, boolean reverse);

    Single<List<Audio>> getDiscography(int offset, int count, boolean reverse);

    Single<List<Photo>> getPhotos(int offset, int count, boolean reverse);

    Single<List<Video>> searchVideos(String q, int offset, int count, boolean reverse);

    Single<List<Audio>> searchAudios(String q, int offset, int count, boolean reverse);

    Single<List<Audio>> searchDiscography(String q, int offset, int count, boolean reverse);

    Single<List<Photo>> searchPhotos(String q, int offset, int count, boolean reverse);

    Single<Integer> update_time(String hash);

    Single<Integer> delete_media(String hash);

    Single<String> get_file_name(String hash);

    Single<Integer> update_file_name(String hash, String name);
}

File: app/src/main/kotlin/dev/ragnarok/fenrir/domain/ILocalServerInteractor.kt
package dev.ragnarok.fenrir.domain

import dev.ragnarok.fenrir.model.Audio
import dev.ragnarok.fenrir.model.Photo
import dev.ragnarok.fenrir.model.Video
import io.reactivex.rxjava3.core.Single

interface ILocalServerInteractor {
    fun getVideos(offset: Int, count: Int, reverse: Boolean): Single<List<Video>>
    fun getAudios(offset: Int, count: Int, reverse: Boolean): Single<List<Audio>>
    fun getDiscography(offset: Int, count: Int, reverse: Boolean): Single<List<Audio>>
    fun getPhotos(offset: Int, count: Int, reverse: Boolean): Single<List<Photo>>
    fun searchVideos(q: String?, offset: Int, count: Int, reverse: Boolean): Single<List<Video>>
    fun searchAudios(q: String?, offset: Int, count: Int, reverse: Boolean): Single<List<Audio>>
    fun searchDiscography(
        q: String?,
        offset: Int,
        count: Int,
        reverse: Boolean
    ): Single<List<Audio>>

    fun searchPhotos(q: String?, offset: Int, count: Int, reverse: Boolean): Single<List<Photo>>
    fun update_time(hash: String?): Single<Int>
    fun delete_media(hash: String?): Single<Int>
    fun get_file_name(hash: String?): Single<String>
    fun update_file_name(hash: String?, name: String?): Single<Int>
}